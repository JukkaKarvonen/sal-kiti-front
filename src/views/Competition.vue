<template>
  <div class="competition">
    <b-row>
      <b-col>
        <h1>{{ $tc("competition.competition", 1) }}: {{ competition.name }}</h1>
      </b-col>
      <b-col v-if="editPermission && $store.state.editMode">
        <b-button
          :to="{
            name: 'competition-update',
            params: {
              event_id: competition.event,
              competition_id: competition.id
            }
          }"
          variant="light"
          class="btn-orange float-right"
          >{{ $tc("competition.edit", 2) }}
        </b-button>
      </b-col>
    </b-row>
    <Competition
      v-on:update:editPermission="editPermission = $event"
      v-on:update:competition="competition = $event"
    />
    <Results
      v-bind:competition="competition"
      v-bind:editPermission="editPermission"
    />
  </div>
</template>

<script>
/**
 * Competition information and results view
 */
import Competition from "@/components/CompetitionInformation.vue";
import Results from "@/components/CompetitionResults.vue";

export default {
  name: "CompetitionView",
  components: {
    Competition,
    Results
  },
  data() {
    return {
      competition: {},
      editPermission: false
    };
  },
  created() {
    document.title =
      this.$t("title.competition") + " | " + this.$t("title.prefix");
  }
};
</script>
