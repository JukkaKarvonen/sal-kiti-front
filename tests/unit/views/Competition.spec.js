import { shallowMount, createLocalVue } from "@vue/test-utils";
import Competition from "@/views/Competition.vue";

import BootstrapVue from "bootstrap-vue";

const localVue = createLocalVue();
localVue.use(BootstrapVue);

describe("Competition.vue", () => {
  const $route = {
    params: { id: 1 }
  };
  it("renders content", () => {
    const wrapper = shallowMount(Competition, {
      localVue,
      mocks: {
        $route
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
