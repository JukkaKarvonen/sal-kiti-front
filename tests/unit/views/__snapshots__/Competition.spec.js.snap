// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Competition.vue renders content 1`] = `
<div
  class="competition"
>
  <b-row-stub
    tag="div"
  >
    <b-col-stub
      tag="div"
    >
      <h1>
        competition.competition: 
      </h1>
    </b-col-stub>
     
    <!---->
  </b-row-stub>
   
  <competition-stub />
   
  <results-stub
    competition="[object Object]"
  />
</div>
`;
